src/compliance/ec2_compliance.py

```diff
@@ -16,35 +16,71 @@ def analyse(self, audit: Audit) -> Set[Findings]:
-                except ValueError:
-                    self.logger.error(
-                        f"image_creation_date '{instance['image_creation_date']}'"
-                        f" for instance '{instance['id']}' does not match the date pattern"
-                        f" YYY-MM-DD[*HH[:MM[:SS[.fff[fff]]]][+HH:MM[:SS[.ffffff]]]]"
-                    )
-                    continue
-
-                image_creation_days_ago = (datetime.now(timezone.utc) - image_creation_date).days
-                if image_creation_days_ago > MAX_AMI_AGE_DAYS:
-                    findings.add(
-                        Findings(
-                            compliance_item_type="ami_creation_age",
-                            account=Account.from_dict(account["account"]),
-                            item=instance["component"],
-                            findings={
-                                f"Instance:  `{instance['id']}`",
-                                f"Component: `{instance['component']}`",
-                                f"Created:   `{image_creation_days_ago}` days ago. ",
-                            },
-                            description=f"Running image AMI created over {MAX_AMI_AGE_DAYS} days ago."
-                            " The older an AMI becomes the higher the likelihood"
-                            " the image is missing important security patches.",
+                    # image_creation_date will be missing if the AMI can no longer be found.
+                    # Here we assume it has been deleted
+                    try:
+                        launch_time = datetime.fromisoformat(instance["launch_time"])
+                    except ValueError:
+                        self.log_date_parse_error(instance["id"], instance["launch_time"])
+                        continue
+                    launch_time_days_ago = (datetime.now(timezone.utc) - launch_time).days
+                    if launch_time_days_ago > MAX_AMI_AGE_DAYS:
+                        findings.add(
+                            self.create_ami_age_finding(
+                                account=account,
+                                instance=instance,
+                                findings={
+                                    f"Instance:  `{instance['id']}`",
+                                    f"Component: `{instance['component']}`",
+                                    f"Started:   `{launch_time_days_ago}` days ago.",
+                                },
+                                description=f"Running image AMI has been deleted"
+                                f" but instance has been running for over {MAX_AMI_AGE_DAYS} days!"
+                                " The older an instance becomes the higher the likelihood"
+                                " the image is missing important security patches.",
+                            )
+                        )
```

Comment:
can be refactored to avoid duplication of code (same logic when using instance date of image date)

Reply:
I have extracted the common parts into create_ami_age_finding(). The parts that remain are the different in text to better explain the finding. 2 lines of the finding set are duplicates, but I have left them here so as to not reduce readability. Please let me know if you have any refactoring ideas

Reply:
I think you can extract it as method and pass a parameter and return the description accordingly, it'll be cleaner with less duplication..

Reply:
We had a chat about this offline, I agree there is still some duplication here but to remove it will add more if statements to the file.